### PR TITLE
Feat/save swap history wallet api exchange

### PR DESCRIPTION
**Pull request by sarneijim**
### 📝 Description

Save swap history wallet api exchange 

### ❓ Context

- **Impacted projects**: LLD
- **Linked resource(s)**:  https://ledgerhq.atlassian.net/browse/LIVE-8943

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx [hunk 1]
```diff
@@ -151,8 +151,7 @@ const SwapForm = () => {
     const { account: fromAccount, parentAccount: fromParentAccount } = from;
     const { account: toAccount, parentAccount: toParentAccount } = to;
     const { feesStrategy } = transaction || {};
-
-    const rateId = exchangeRate?.rateId || "12345";
+    const { rate, rateId = "1234" } = exchangeRate || {};
```
Comment (kallen-ledger) [inline apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.tsx:154]: Do we want the fallback of "1234" here?

**Review by sprohaszka-ledger — COMMENTED** [review]
I think we could put elsewhere the `ExchangeDrawer/utils.tsx` file.
It returns a React Action that should also be used later by Ledger Mobile. Maybe Desktop and Mobile doesn't share the exact way to manage this action, otherwise we should consider putting it inside `live-common`.

All those logical function shouldn't be in a tsx file or in LLM nor LLD.

File: apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/ExchangeDrawer/utils.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import { addPendingOperation } from "@ledgerhq/live-common/account/index";
+import { getMainAccount } from "@ledgerhq/live-common/account/helpers";
+import { getEnv } from "@ledgerhq/live-env";
+import { postSwapAccepted, postSwapCancelled } from "@ledgerhq/live-common/exchange/swap/index";
+import addToSwapHistory from "@ledgerhq/live-common/exchange/swap/addToSwapHistory";
+import { Account, Operation } from "@ledgerhq/types-live";
+import { Transaction } from "@ledgerhq/live-common/generated/types";
+import { Exchange as SwapExchange, ExchangeRate } from "@ledgerhq/live-common/exchange/swap/types";
+
+export const onCompleteExchange = ({
```
[the remark this answers is not in the document]
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/ExchangeDrawer/utils.tsx]: Renamed to "getUpdateAccountActionAfterSwap". Open to change the name if you find a better one.

File: apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx [hunk 3]
```diff
@@ -56,11 +69,42 @@ const Body = ({ data, onClose }: { data: Data; onClose?: () => void | undefined
   useEffect(() => {
     if (signedOperation) {
       broadcast(signedOperation).then(operation => {
+        // Save swap history
+        if (swapId && rate && toAccount) {
+          const result = { operation, swapId };
+          const unitFrom = getAccountUnit(account);
+          const unitTo = getAccountUnit(toAccount);
+          const magnitudeAwareRate = new BigNumber(rate).div(
```
[the remark this answers is not in the document]
Comment (sarneijim) [reply apps/ledger-live-desktop/src/renderer/modals/Platform/Exchange/CompleteExchange/Body.tsx]: Done. Created convertParametersToValidFormat function.

File: pnpm-lock.yaml [hunk 4]
```diff
@@ -28053,7 +28053,7 @@ packages:
       babel-plugin-syntax-jsx: 6.18.0
       lodash: 4.17.21
       picomatch: 2.3.1
-      styled-components: 5.3.5(react-is@18.1.0)(react@18.0.0)
+      styled-components: 5.3.5(react-dom@17.0.2)(react-is@17.0.2)(react@17.0.2)
```
Comment (sarneijim) [inline pnpm-lock.yaml:28056]:
```suggestion
      styled-components: 5.3.5(react-is@18.1.0)(react@18.0.0)
```

**Review by sprohaszka-ledger — COMMENTED** [review]
I have only one doubt left: why do we have twice the call to `updateAccountWithUpdater` and the dispatch of the AccountUpdater. Don't it will be called twice?

**Comment by sarneijim** [comment]
> I have only one doubt left: why do we have twice the call to `updateAccountWithUpdater` and the dispatch of the AccountUpdater. Don't it will be called twice?

- getUpdateAccountWithUpdaterParams return the params used in react action
- updateAccountWithUpdater is creating the action.
export const updateAccountWithUpdater = createAction<AccountsUpdateAccountWithUpdaterPayload>(
  AccountsActionTypes.UPDATE_ACCOUNT,
);
- dispatch execute it

Is this your question?

**Review by sprohaszka-ledger — APPROVED** [review]

**Pull request merged**